presto-main/src/main/java/com/facebook/presto/operator/aggregation/TruncatedHistogramState.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation;
+
+import com.facebook.presto.operator.aggregation.state.AbstractGroupedAccumulatorState;
+import com.facebook.presto.operator.aggregation.state.AccumulatorState;
+import com.facebook.presto.operator.aggregation.state.AccumulatorStateFactory;
+import com.facebook.presto.operator.aggregation.state.AccumulatorStateMetadata;
+import com.facebook.presto.operator.aggregation.state.AccumulatorStateSerializer;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.type.MapType;
+import com.facebook.presto.type.RowType;
+import com.facebook.presto.util.array.ObjectBigArray;
+import com.google.common.collect.ImmutableList;
+
+import java.util.Optional;
+
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static java.util.Objects.requireNonNull;
+
+@AccumulatorStateMetadata(stateSerializerClass = TruncatedHistogramState.Serializer.class, stateFactoryClass = TruncatedHistogramState.Factory.class)
+public interface TruncatedHistogramState
+        extends AccumulatorState
+{
+    void set(TypedTruncatedHistogram histogram);
+
+    TypedTruncatedHistogram get();
+
+    void addMemoryUsage(long memory);
+
+    class Factory
```

Comment:
Follow the convention of other state interfaces, and make the factory and serializer top-level classes